#!/usr/bin/env python2
PACKAGE = "cam_sync"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("fps", double_t,0,"fps setting for master camera", 60, 0.1, 160);
gen.add("max_free_fps", double_t,0,"maximum fps of free running camera (don't change!)", 200, 1, 1000);

gen.add("shutter_ms", double_t,0,"shutter_ms", 0.0001, 0, 500.0);
gen.add("gain_db", double_t,0,"gain_db", 0, 0, 30.0);

gen.add("white_balance", bool_t, 0, "Enable/disable white balance", False)
gen.add("auto_white_balance", bool_t, 0, "Automatically change white balance", True)
gen.add("wb_blue", int_t, 0, "White balance blue component", 800, 0, 1023)
gen.add("wb_red", int_t, 0, "White balance red component", 550, 0, 1023)

gen.add("trigger_source", int_t, 0, "trigger source (GPIO pin)", -1, -1, 4);
gen.add("trigger_polarity", int_t, 0, "trigger polarity (0 or 1)", 0, 0, 1);
gen.add("strobe_control", int_t, 0, "trigger source (GPIO pin)", 2,  -1, 4);
gen.add("strobe_polarity", int_t, 0, "strobe polarity (0 or 1)", 0, 0, 1);

gen.add("width", int_t,0,"width", 2048, 0, 2048);
gen.add("height", int_t,0,"height", 2048, 0, 2048);
gen.add("format7_mode", int_t, 0, "Camera modes for format 7", 0, 0, 8)

exit(gen.generate(PACKAGE, "cam_sync", "CamSyncDyn"))
